.github/workflows/build.yaml

```diff
@@ -15,13 +15,46 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - uses: actions/checkout@v3
+      - name: Identify week number
+        shell: bash
+        run: |
+          date +'%V' > week-number.txt
+          date +'week-number=%V' >> $GITHUB_ENV
       - uses: actions/setup-python@v4
+        id: setup-python
         with:
           python-version: "3.10"
+          cache: "pip"
+          cache-dependency-path: |
+            .github/workflows/build.yaml
+            setup.cfg
+            setup.py
+            tox.ini
+            week-number.txt
+      - name: Restore cache
+        id: restore-cache
+        uses: actions/cache@v3
+        with:
+          path: |
+            .tox/
+            .venv/
+          key: >
+            lint
+            week=${{ env.week-number }}
+            os=${{ matrix.os }}
+            python=${{ steps.setup-python.outputs.python-version }}
+            hash=${{ hashFiles('.github/workflows/build.yaml', 'tox.ini', 'setup.cfg', 'setup.py') }}
```

Comment:
Tangent: let's switch to declarative `setup.cfg` package metadata and make `setup.py` just a call to `setup()`, so we can remove `setup.py` from these file hash lists! 😉